simple_history/management/commands/clean_duplicate_history.py

```diff
@@ -115,6 +119,8 @@ def _process_instance(self, instance, model, stop_date=None, dry_run=True):
             for f2 in o_qs[1:]:
                 entries_deleted += self._check_and_delete(f1, f2, dry_run)
                 f1 = f2
+                if entries_deleted > 0 and entries_deleted % batch_size == 0:
```

Comment:
Is this form of batching good enough or is it important that we also break up the transaction?

Reply:
Yeah I think we need to break up the transaction. I think this cron is a major offender when we see huge oldest transaction times coming from `readwrite_cron` as this process takes many minutes.

In order to really break up the transaction, we'd need to be able to get rid of the `with transaction.atomic()` wrapping all of this. This should be a separate option since I'm not sure everyone will want to do that but I think for us, we see it as an idempotent operation and we do not want to revert on failure, especially since the reversion could be huge.

Reply:
Added this.